modules/stochastic_tools/src/utils/GaussianProcess.C

```diff
@@ -147,12 +152,215 @@ GaussianProcess::standardizeData(RealEigenMatrix & data, bool keep_moments)
     _data_standardizer.computeSet(data);
   _data_standardizer.getStandardized(data);
 }
+                                    
+void
+GaussianProcess::logl(const RealEigenMatrix & out_vec, const RealEigenMatrix & x1, const RealEigenMatrix & x2, Real g, const RealEigenMatrix & theta, 
+          LogLResult & result, bool outer, bool tau2) {
+  int n = out_vec.rows();
+  RealEigenMatrix K(x1.rows(), x2.rows());
+
+  std::vector<Real> theta1(_num_tunable, 0.0);
```

Comment:
We have an automated way of dealing with this through the hyper parameter maps. The scaling parameter might not be at the same location all the time. So I'd have a loop checking for it.